File: src/main/java/ley/anvil/modpacktools/command/CommandReturn.java
package ley.anvil.modpacktools.command;

public class CommandReturn {
    private final String ret;
    private final boolean success;

    private CommandReturn(String ret, boolean success) {
        this.ret = ret;
        this.success = success;
    }

    public static CommandReturn fail(String ret) {
        return new CommandReturn(ret, false);
    }

    public static CommandReturn success() {
        return success("");
    }

    public static CommandReturn success(String ret) {
        return  new CommandReturn(ret, true);
    }

    public boolean hadSuccess() {
        return success;
    }

    public String getRet() {
        return ret;
    }
}

File: src/main/kotlin/ley/anvil/modpacktools/command/CommandReturn.kt
package ley.anvil.modpacktools.command

data class CommandReturn constructor(val ret: String?, val success: Boolean) {
    companion object {
        /**
         * Get a failed [CommandReturn]. This should be used if something went wrong
         *
         * @param ret the error message
         * @return the [CommandReturn]
         */
        @JvmStatic
        @JvmOverloads
        fun fail(ret: String? = null) = CommandReturn(ret, false)

        /**
         * Get a successful [CommandReturn]. Use this if the command was executed successfully
         *
         * @param ret a return message
         * @return the [CommandReturn]
         */
        @JvmStatic
        @JvmOverloads
        fun success(ret: String? = null) = CommandReturn(ret, true)
    }
    fun hasRet() = ret != null
}
